bbot/modules/deadly/medusa.py

```diff
@@ -0,0 +1,233 @@
+import re
+from bbot.modules.base import BaseModule
+from bbot.errors import WordlistError
+
+
+class medusa(BaseModule):
+    watched_events = ["PROTOCOL"]
+    produced_events = ["VULNERABILITY"]
+    flags = ["active", "aggressive", "deadly"]
+    per_host_only = True
+    meta = {
+        "description": "Medusa SNMP bruteforcing with v1, v2c and R/W check.",
+        "created_date": "2025-05-16",
+        "author": "@christianfl",
+    }
+    scope_distance_modifier = None
+
+    options = {
+        "snmp_wordlist": "https://raw.githubusercontent.com/danielmiessler/SecLists/refs/heads/master/Discovery/SNMP/snmp.txt",
+        "snmp_versions": ["1", "2C"],  # Only 1 and 2C are available with medusa 2.3.
+        "wait_ms": 500,
+        "timeout_s": 30,
+    }
+
+    options_desc = {
+        "snmp_wordlist": "Wordlist url for SNMP community strings, newline separated (default https://raw.githubusercontent.com/danielmiessler/SecLists/refs/heads/master/Discovery/SNMP/snmp.txt)",
+        "snmp_versions": "List of SNMP versions to attempt against the SNMP server (default ['1', '2C'])",
+        "wait_ms": "Wait time after every SNMP request in milliseconds (default 500)",
+        "timeout_s": "Wait time for the SNMP response(s) once at the end of all attempts (default 30)",
+    }
```

Comment:
There are a few changes we may want to make here, to speed up the module:

- decrease the default `wait_ms` to `0`, unless needed for typical SNMP setups
- add a `threads` option (`-t`), with a default of 5 or more
- by default, use a smaller wordlist (instead of 3000 entries, something closer to 50, e.g. only the most common default values like `admin`, `cisco`, `public`, etc.)

Let me know your thoughts. The main goal is to speed up brute forcing so that each host completes in 10 seconds or less, with the option to use a larger wordlist if needed.

Reply:
- Changed wait_ms to wait_microseconds and use 200 (default Medusa SNMP value). Is it okay like that? I never tested a lower value here
- Added threads option
- Use wordlist with 118 entries. Please ping if that's too large. It's a very common one I think.

Generally, I aim to not scan too fast on the external side but as long as everything has options, I'm totally fine with the changes.